src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/ViewCssScopePass.cs

```diff
@@ -53,12 +54,31 @@ private void ProcessElement(HtmlContentIntermediateNode node, string cssScope)
                     });
                     i++;
                 }
+
+                previousTokenOpt = token;
             }
         }
 
-        static bool IsValidElement(IntermediateToken token)
+        static bool IsValidElement(IntermediateToken token, IntermediateToken previousTokenOpt)
         {
-            var content = token.Content;
+            var content = token.Content.AsSpan();
+
+            // `<!tag` is lowered into separate nodes `<` and `tag`, we process the latter.
+            if (previousTokenOpt?.Content == "<" && !content.StartsWith("<".AsSpan(), StringComparison.Ordinal) && content.Length != 0)
```

Comment:
```suggestion
            if (previousTokenOpt?.Content == "<" && content is [not '<', ..])
```

Reply:
Thanks. Simplified also the other `if` below similarly.